src/Lean/Meta/Transform.lean

```diff
@@ -199,6 +202,37 @@ def unfoldDeclsFrom (biggerEnv : Environment) (e : Expr) : CoreM Expr := do
         return .done e
     Core.transform e (pre := pre)
 
+/--
+Unfolds theorem applied to `.const n` where `n` in the given array.
+This is used to undo proof abstraction for termination checking, as otherwise the bare occurence of
+the recursive functions prevents termination checking.
+Unfolds from the private environment.
```

Comment:
```suggestion
Unfolds theorems that are applied to a `.const n` where `n` in the given array.
This is used to undo proof abstraction for termination checking, as otherwise the bare
occurrence of the recursive function prevents termination checking from succeeding.

This unfolds from the private environment. The resulting definitions are (usually) not
exposed anyways.
```